frontend/lib/src/components/shared/InputInstructions/InputInstructions.tsx

```diff
@@ -54,7 +54,7 @@ const InputInstructions = ({
   }
 
   // Show enter instruction if not a form or form allows Enter to submit
-  if (dirty && allowEnterToSubmit) {
+  if ((inForm || dirty) && allowEnterToSubmit) {
```

Comment:
Think this is a good opportunity to simplify the logic. We are concerned about 2 scenarios here, as far as I understand:
1) If in a form & Enter to Submit is allowed 
**OR**
2) Not in a form & dirty state

_(since focused is handled in the parent component)_

I would propose handling this in the prop `allowEnterToSubmit` instead - see line 534 of `NumberInput.tsx`. 
```
allowEnterToSubmit={allowFormEnterToSubmit || (!inForm && dirty) }
```

Reply:
Yep, I agree. I felt that the various states and booleans are getting cumbersome.